lib/src/query_component.dart

```diff
@@ -73,18 +73,17 @@ class QueryComponent implements OnInit {
         ? _filterInFlightInsertionOrders(insertionOrders)
         : insertionOrders;
 
-    // TODO: update reporting query to deal with multiple IOs.
-    // Issue: https://github.com/googleinterns/dv360-excel-plugin/issues/61
-    final insertionOrder = insertionOrders.first;
+    // Gets the earliest start date from the list of insertion orders.
+    final minStartDate = _getMinStartDate(insertionOrders);
 
-    // Gets dateRange for the active budget segment.
-    final activeDateRange = insertionOrder.budget.activeBudgetSegment.dateRange;
+    // Uses DBM reporting APIs to get spent (currency based or
+    // impression based) within time window [minStartDate, Now].
+    // spendingMap has a [DailySpend] for each insertion order.
+    final spendingMap =
+        await _queryAndParseSpentData(minStartDate, DateTime.now());
 
-    // Uses DBM reporting APIs to get revenue data within [activeDateRange].
-    final revenueMap = await _queryAndParseRevenueSpentData(activeDateRange);
-
-    // Add revenue spent to the parsed Insertion Order.
-    insertionOrder.spent = revenueMap[insertionOrder.insertionOrderId] ?? '';
+    // Add spent to the list of insertion orders.
+    _addSpentToInsertionOrders(insertionOrders, spendingMap);
```

Comment:
i like this method name because it makes it clear the operation is in place. nice